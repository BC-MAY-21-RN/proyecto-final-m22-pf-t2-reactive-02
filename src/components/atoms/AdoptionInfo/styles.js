import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  title: {
    fontSize: 23,
    fontWeight: 'bold',
    marginVertical: 20,
    alignSelf: 'center',
  },
});

export default styles;
