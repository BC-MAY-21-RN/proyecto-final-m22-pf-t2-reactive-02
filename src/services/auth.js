import auth from '@react-native-firebase/auth';
import {GoogleSignin} from '@react-native-google-signin/google-signin';

function forgetPassword(text) {
  return auth().sendPasswordResetEmail(text);
}

function getId() {
  return auth().currentUser.uid;
}

function getName() {
  return auth().currentUser.displayName;
}

function getEmail() {
  return auth().currentUser.email;
}

function getPhoto() {
  return auth().currentUser.photoURL;
}

async function loginAccount(email, password) {
  return auth().signInWithEmailAndPassword(email, password);
}

function logout() {
  auth()
    .signOut()
    .then(() => {});
}

async function registerEmailUser(user) {
  return auth().createUserWithEmailAndPassword(
    user.object.email,
    user.object.password,
  );
}

async function onGoogleButtonPress() {
  const {idToken} = await GoogleSignin.signIn();
  const googleCredential = auth.GoogleAuthProvider.credential(idToken);
  return auth().signInWithCredential(googleCredential);
}

function updateUser(user, getPhotoUrl) {
  const update = {
    displayName: user.object.name,
    photoURL: getPhotoUrl(auth().currentUser.photoURL),
  };
  auth().currentUser.updateProfile(update);
}

export default {
  forgetPassword,
  getId,
  getName,
  getEmail,
  getPhoto,
  loginAccount,
  logout,
  onGoogleButtonPress,
  registerEmailUser,
  updateUser,
};
