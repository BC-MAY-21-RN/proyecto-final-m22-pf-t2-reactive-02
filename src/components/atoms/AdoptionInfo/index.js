import React from 'react';
import {Text, View} from 'react-native';
import styles from './styles';
import TwoTexts from '../TwoTexts';

export default function AdoptionInfo({data}) {
  return (
    <View>
      <Text style={styles.title}>Respuestas del usuario</Text>
      <TwoTexts text={'Nombre: '} text2={data.nombreUsuario} styles={styles} />
      <TwoTexts text={'Telefono: '} text2={data.telefono} styles={styles} />
      <TwoTexts text={'Correo: '} text2={data.correo} styles={styles} />
      <TwoTexts text={'Ciudad: '} text2={data.ciudad} styles={styles} />
      <Text style={styles.text}>¿Cuántas mascotas tienes?</Text>
      <Text>{data.mascotas}</Text>
      <Text style={styles.text}>
        ¿Conviven con otras mascotas de forma pacifica?
      </Text>
      <Text>{data.Pacifico}</Text>
      <Text style={styles.text}>
        ¿Cuántas horas al día pasará tu mascota sola?{' '}
      </Text>
      <Text>{data.horas}</Text>
      <Text style={styles.text}>Tu mascota estará mayormente... </Text>
      <Text>{data.lugarMascota}</Text>
    </View>
  );
}
