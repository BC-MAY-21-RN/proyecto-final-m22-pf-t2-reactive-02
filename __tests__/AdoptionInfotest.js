import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AdoptionInfo from '../src/components/atoms/AdoptionInfo';
import '@react-native-firebase/auth';
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
jest.mock('@react-native-google-signin/google-signin', () => {});

describe('Test from AdoptionInfo component', () => {
  it('AdoptionInfo snapshot test', () => {
    const renderAdoptionInfo = renderer
      .create(<AdoptionInfo data={{}} key={1} />)
      .toJSON();
    expect(renderAdoptionInfo).toMatchSnapshot();
  });
});
