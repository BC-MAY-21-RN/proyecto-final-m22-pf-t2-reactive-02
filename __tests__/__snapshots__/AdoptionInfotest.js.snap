// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from AdoptionInfo component AdoptionInfo snapshot test 1`] = `
<View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 23,
        "fontWeight": "bold",
        "marginVertical": 15,
      }
    }
  >
    Respuestas del usuario
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 17,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      Nombre: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 15,
          "marginLeft": 10,
          "maxWidth": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 17,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      Telefono: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 15,
          "marginLeft": 10,
          "maxWidth": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 17,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      Correo: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 15,
          "marginLeft": 10,
          "maxWidth": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 17,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      Ciudad: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 15,
          "marginLeft": 10,
          "maxWidth": 150,
          "width": 150,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
        "paddingTop": 10,
      }
    }
  >
    ¿Cuántas mascotas tienes?
  </Text>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
        "paddingTop": 10,
      }
    }
  >
    ¿Conviven con otras mascotas de forma pacifica?
  </Text>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
        "paddingTop": 10,
      }
    }
  >
    ¿Cuántas horas al día pasará tu mascota sola?
     
  </Text>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
        "paddingTop": 10,
      }
    }
  >
    Tu mascota estará mayormente... 
  </Text>
  <Text />
</View>
`;
